lib/action_policy/i18n.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module ActionPolicy
+  module I18n # :nodoc:
+    class << self
+      def full_message(policy_class, rule)
+        unless defined?(::I18n)
+          raise "I18n is not available! Please make sure you have 'i18n' gem installed"
+        end
+
+        candidates = candidates_for(policy_class, rule)
+
+        ::I18n.t(
+          candidates.shift,
+          default: candidates,
+          scope: [:action_policy, :policy]
+        )
+      end
+
+      private
+
+      def candidates_for(policy_class, rule)
+        policy_hierarchy = policy_class.ancestors.select { |klass| klass < ActionPolicy::Base }
```

Comment:
Calling `policy_class` is not very efficient: it could be a long list of all modules and class up to BaseObject.

See, for example, how [ActiveRecord performs lookup](https://github.com/rails/rails/blob/b2eb1d1c55a59fee1e6c4cba7030d8ceb524267c/activerecord/lib/active_record/translation.rb#L8).

`klass < ActionPolicy::Base` – this check won't work if we do not inherit from `ActionPolicy::Base`

ActionPolicy allows to build a custom _base_ policy.

The only requirement for policy class is to include `ActionPolicy::Policy::Core` module.
We can check, for example, whether class `respond_to?(:identifier)` or not. Duck typing)

Reply:
✔️